
# EE282 HW#3


### Loading modules
```
module load perl
module load R
module load jje/kent/2014.02.19
module load jje/jjeutils/0.1a
```
### Downloading the all chromosomes fasta file and md5sum and check file integrity 

```
mkdir genome
cd genome
wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/fasta/dmel-all-chromosome-r6.24.fasta.gz

wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/fasta/md5sum.txt

grep -P '.+all-chromosome.+' md5sum.txt >> md5new.txt
md5sum -c md5new.txt
md5sum *.gz > md5check.txt
diff md5{new,check}.txt

```
If there is no output your file intergrity is good!

### Calculating the total number of nucleotides, Ns, and Sequences 
```
faSize dmel-all-chromosome-r6.24.fasta.gz > frequency.txt

```
### Downloading the .gtf file, and checking file integrity

```
cd ../
mkdir annotation
cd annotation

wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/gtf/*

md5sum -c md5sum.txt
md5sum *.gz > md5check.txt
diff md5{sum,check}.txt
```
If there is no output your file intergrity is good!

### Counting the number of features of each type
```
zcat dmel-all-r6.24.gtf.gz \
| awk '{print $3}'\
| sort \
| uniq -c \
| sort -rn > Summary.txt
less Summary.txt

```
### Counting the genes on each chromosomes
```
zcat dmel-all-r6.24.gtf.gz \
| awk '{print $1"\t"$3}' \
| grep gene \
| awk '{print $1}' \
| sort \
| uniq -c \
| sort -rn \
| head -7 > Genes_per_chromosome.txt
less Genes_per_chromosome.txt
```

### Comments

Good job.

```md5sum -c``` is very useful. Try the following (provided you maintained the filenames):

```
$ wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/gtf/md5sum.txt
$ wget ftp://ftp.flybase.net/genomes/Drosophila_melanogaster/current/gtf/dmel-all-r6.24.gtf.gz
$ md5sum dmel-all-r6.24.gtf.gz 
5cd5dcfbfff952ea7ce89e26cba89bbd  dmel-all-r6.24.gtf.gz

$ md5sum -c md5sum.txt
dmel-all-r6.24.gtf.gz: OK
```
